src/main/java/bio/terra/cda/app/controller/QueryApiController.java

```diff
@@ -95,24 +94,21 @@ private String createNextUrl(String jobId, int offset, int limit) {
     return sendQuery(querySql, dryRun);
   }
 
+  /**
+   * Table cells that represent as nested structures must be unnested from it decendant heiarchy.
+   * the UNNEST clause is generated by unwinding the "dot" hierarchy such that:
+   * gdc-bg-sample.cda_mvp.v3.ResearchSubject.identifier.value would unwind to the following.
+   * "SELECT DISTINCT (_D.value) FROM gdc-bq-sample.cda_mvp.v3, UNNEST(ResearchSubject) as _C,
+   * UNNEST(_C.identifier) AS _D";
+   */
   @Override
-  public ResponseEntity<QueryResponseData> uniqueValues(String version, @Valid String body) {
+  public ResponseEntity<QueryCreatedData> uniqueValues(String version, String body) {
 
-    String bqTable = "gdc-bq-sample";
-    Map<String, String> tableParts = QueryTranslator.parseTableName(body);
-    // SELECT * FROM UNNEST (SELECT DISTINCT p.column FROM qualifiedTABLE as p);
-    String querySql =
-        "SELECT DISTINCT "
-            + tableParts.get("columnName")
-            + " FROM "
-            + bqTable
-            + "."
-            + tableParts.get("tableName");
-    logger.info("uniqueValues: " + querySql);
-
-    ResponseEntity<QueryCreatedData> response = sendQuery(querySql, false);
-    var query_id = response.getBody().getQueryId();
-
-    return query(query_id, 0, 1000);
+    String table = applicationConfiguration.getBqTable() + "." + version;
+    NestedColumn nt = new bio.terra.cda.app.model.NestedColumn(body);
```

Comment:
Since this class is imported you don't need to use a fully qualified name.
```suggestion
    NestedColumn nt = new NestedColumn(body);
```

Reply:
IDE did that...fixed.